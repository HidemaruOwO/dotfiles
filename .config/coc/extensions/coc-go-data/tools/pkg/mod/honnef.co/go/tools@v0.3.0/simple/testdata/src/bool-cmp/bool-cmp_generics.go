//go:build go1.18

package pkg

func tpfn1[T any]() T { var zero T; return zero }

func tpfn() {
	if tpfn1[bool]() == true { // want `simplified to tpfn1\[bool\]\(\)`
	}
	if tpfn1[any]() == true {
	}
}

func tpfn2[T bool](x T) {
	if x == true { // want `omit comparison to bool constant`
	}
}

func tpfn3[T ~bool](x T) {
	if x == true { // want `omit comparison to bool constant`
	}
}

type MyBool bool

func tpfn4[T bool | MyBool](x T) {
	if x == true { // want `omit comparison to bool constant`
	}
}
