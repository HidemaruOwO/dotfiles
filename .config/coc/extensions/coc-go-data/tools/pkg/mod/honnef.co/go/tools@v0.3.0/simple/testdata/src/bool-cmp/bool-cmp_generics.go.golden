//go:build go1.18

package pkg

func tpfn1[T any]() T { var zero T; return zero }

func tpfn() {
	if tpfn1[bool]() { // want `simplified to tpfn1\[bool\]\(\)`
	}
	if tpfn1[any]() == true {
	}
}

func tpfn2[T bool](x T) {
	if x { // want `omit comparison to bool constant`
	}
}

func tpfn3[T ~bool](x T) {
	if x { // want `omit comparison to bool constant`
	}
}

type MyBool bool

func tpfn4[T bool | MyBool](x T) {
	if x { // want `omit comparison to bool constant`
	}
}
